src/main/java/spireMapOverhaul/SpireAnniversary6Mod.java

```diff
@@ -483,6 +503,9 @@ else if (Modifier.isStatic(m.getModifiers()) && m.getName().equals("endsWithRewa
                     }
                     eventBuilder.endsWithRewardsUI(endsWithRewardsUI);
 
+                    if (eventID.equals(Surprise.ID))
```

Comment:
A new way to do this was recently added. You can delete this and instead add the following method to your event class:
```java
    public static boolean endsWithRewardsUI() {
        return true;
    }
```